Update Ingestion Sampling with OTel Documentation (#22773)
DataDog/documentation: content/en/opentelemetry/ingestion_sampling_with_opentelemetry.md

## Overview

OpenTelemetry SDKs and the OpenTelemetry Collector provide sampling capabilities, as ingesting 100% of traces is often unnecessary to gain visibility into the health of your applications. Configure sampling rates before sending traces to Datadog to ingest data that is most relevant to your business and observability goals, while controlling and managing overall costs.

This document demonstrates two primary methods for sending traces to Datadog with OpenTelemetry:

- Send traces to the **[OpenTelemetry Collector][1]**, and use the Datadog Exporter to forward them to Datadog.
- Send traces to the **[Datadog Agent OTLP ingest][3]**, which forwards them to Datadog.

**Note**: Datadog doesn't support running the OpenTelemetry Collector and the Datadog Agent on the same host.

### Using the OpenTelemetry Collector

> With this method, the OpenTelemetry Collector receives traces from OpenTelemetry SDKs and exports them to Datadog using the Datadog Exporter. Here, the Datadog Connector computes [APM trace metrics][4]:

Author: Good suggestion. I agree with your points!

## Reducing ingestion volume

With OpenTelemetry, you can configure sampling both in the OpenTelemetry libraries and in the OpenTelemetry collector: 

- **Head-based sampling** in the OpenTelemetry SDKs
- **Tail-based sampling** in the OpenTelemetry Collector
- **Probabilistic sampling** in the Datadog Agent

### Head-based sampling

At the SDK level, you can implement _head-based sampling_. This is when the sampling decision is made at the beginning of the trace. This type of sampling is particularly useful for high-throughput applications, when you have a clear understanding of which traces are most important to ingest and want to make sampling decisions early in the tracing process.

#### Configuring

To configure head-based sampling, use the [TraceIdRatioBased][5] or [ParentBased][6] samplers provided by the OpenTelemetry SDKs. These allow you to implement deterministic head-based sampling based on the `trace_id` at the SDK level.

#### Considerations

Head-based sampling affects the computation of APM metrics because only the sampled traces are sent to the OpenTelemetry Collector or Datadog Agent, and this is where the metrics calculation is done.

> To get accurate metrics, upscale them using [formulas and functions][7] in Datadog dashboards and monitors, provided that you know the configured sampling rate in the SDK.

Reviewer: ```suggestion
To approximate real metrics from sampled metrics, use [formulas and functions][7] with the sampling rate configured in the SDK.
```
- phrasing it as "approximate" because that's sort of the best you can do with upscaling
- "real" may not be the best word here, I've just suggested it as a placeholder. is there a better way to describe "actual" (that is, not sampled) metrics? 
- I'm not sure about the use of the word "upscale" here. I've definitely seen it used at Datadog to talk about going from sampled data to real/actual data—but outside of DD, I usually see it in the context of AI models. Really not sure if customers will understand the meaning in the way we want them to

Author: Great points and suggestions. I think instead of real metrics, we could probably just say "unsampled metrics".

Reviewer: oooh "unsampled metrics" is perfect!

Use the [ingestion volume control guide][8] to read more about the implications of setting up trace sampling on trace analytics monitors and metrics from spans.

### Tail-based sampling

> At the OpenTelemetry Collector level, you can do _tail-based sampling_, which allows you to define more advanced rules to keep accrued visibility over error or high latency traces.

Reviewer: ```suggestion
At the OpenTelemetry Collector level, you can do _tail-based sampling_, which allows you to define more advanced rules to keep accrued visibility over errors or high latency traces.
```

could use a confidence check on whether i'm parsing this sentence correctly... should it actually be "visibility over error traces or high latency traces"?

Author: Good thinking. I think we could clarify that with something like:
```suggestion
At the OpenTelemetry Collector level, you can do _tail-based sampling_, which allows you to define more advanced rules to maintain visibility over traces with errors or high latency.
```

#### Configuring

> To configure tail-based sampling, use the [Tail Sampling Processor][9] and [Probabilistic Sampling Processor][10] to sample traces based on a set of rules at the collector level.

Reviewer: ```suggestion
To configure tail-based sampling, use the [Tail Sampling Processor][9] or [Probabilistic Sampling Processor][10] to sample traces based on a set of rules at the collector level.
```

#### Considerations

> Tail sampling's main limitation is that all spans for a given trace must be received by the same collector instance for effective sampling decisions. If the trace is distributed across multiple collector instances, there's a risk that some parts of a trace are dropped whereas some other parts of the same trace are sent to Datadog.

Reviewer: ```suggestion
A limitation of tail-based sampling is that all spans for a given trace must be received by the same collector instance for effective sampling decisions. If a trace is distributed across multiple collector instances, and tail-based sampling is used, some parts of that trace may not be sent to Datadog.
```

See the [ingestion volume control guide][8] for information about the implications of setting up trace sampling on trace analytics monitors and metrics from spans.

### Probabilistic sampling

When using Datadog Agent OTLP ingest, a probabilistic sampler is available starting with Agent v7.44.0. 

#### Configuring

To configure probabilistic sampling, either:

- Set the `DD_OTLP_CONFIG_TRACES_PROBABILISTIC_SAMPLER_SAMPLING_PERCENTAGE` environment variable, or
- Add the following YAML to your Agent's configuration file:

  ```yaml
  otlp_config:
    # ...
    traces:
      probabilistic_sampler:
        sampling_percentage: 50 #In this example, 50% of traces are captured.
  ```
#### Considerations

The probabilistic sampler ignores spans for which the sampling priority is set at the SDK level. Spans not captured by the probabilistic sampler may still be captured by the Datadog Agent's [error and rare samplers][12].

> ## Monitoring ingested volumes from the Datadog UI

Reviewer: ```suggestion
## Monitoring ingested volumes in Datadog
```
